src/pages/[platform]/build-a-backend/auth/connect-your-frontend/switching-authentication-flows/index.mdx

```diff
@@ -274,6 +275,387 @@ To create a CAPTCHA challenge with a Lambda Trigger, please visit [AWS Amplify G
+
+When you have migrated all your users, switch flows to the more secure SRP flow. The SRP flow doesn't send any passwords over the network.
+
+<BlockSwitcher>
+<Block name="Java">
+
+```java
+AuthSignInOptions options = AWSCognitoAuthSignInOptions.builder()
+    .authFlowType(AuthFlowType.USER_PASSWORD_AUTH)
+    .build();
+Amplify.Auth.signIn(
+    "username",
+    "password",
+    options,
+    result -> Log.i("AuthQuickStart", "Sign in succeeded with result " + result),
+    error -> Log.e("AuthQuickStart", "Failed to sign in", error)
+);
+```
+
+</Block>
+<Block name="Kotlin - Callbacks">
+
+```kotlin
+val options = AWSCognitoAuthSignInOptions.builder()
+    .authFlowType(AuthFlowType.USER_PASSWORD_AUTH)
+    .build()
+Amplify.Auth.signIn(
+    "username",
+    "password",
+    options,
+    { result ->
+        Log.i("AuthQuickstart", "Next step for sign in is ${result.nextStep}")
+    },
+    { error ->
+        Log.e("AuthQuickstart", "Failed to sign in", error)
+    }
+)
+```
+
+</Block>
+<Block name="Kotlin - Coroutines">
+
+```kotlin
+try {
+    val options = AWSCognitoAuthSignInOptions.builder()
+        .authFlowType(AuthFlowType.USER_PASSWORD_AUTH)
+        .build()
+    val result = Amplify.Auth.signIn(
+        username = "username",
+        password = "password",
+        options = options
+    )
+    Log.i("AuthQuickstart", "Next step for sign in is ${result.nextStep}")
+} catch (error: AuthException) {
+    Log.e("AuthQuickstart", "Sign in failed", error)
+}
+```
+
+</Block>
+<Block name="RxJava">
+
+```java
+AuthSignInOptions options = AWSCognitoAuthSignInOptions.builder()
+    .authFlowType(AuthFlowType.USER_PASSWORD_AUTH)
+    .build();
+RxAmplify.Auth.signIn("username", "password", options)
+    .subscribe(
+        result -> Log.i("AuthQuickstart", "Next step for sign in is " + result.getNextStep()),
+        error -> Log.e("AuthQuickstart", error.toString())
+    );
+```
+
+</Block>
+</BlockSwitcher>
+
+### Set up auth backend
+
+In order to use the authentication flow `USER_PASSWORD_AUTH`, your Cognito app client has to be configured to allow it. Amplify Gen 2 enables SRP auth by default. To enable USER_PASSWORD_AUTH, you can update the `backend.ts` file with the following changes:
+
+```ts title="amplify/backend.ts"
+import { defineBackend } from '@aws-amplify/backend'
+import { auth } from './auth/resource'
+import { data } from './data/resource'
+
+const backend = defineBackend({
+  auth,
+  data,
+});
+
+// highlight-start
+backend.auth.resources.cfnResources.cfnUserPoolClient.explicitAuthFlows = [
+  "ALLOW_USER_PASSWORD_AUTH",
+  "ALLOW_USER_SRP_AUTH",
+  "ALLOW_USER_AUTH",
+  "ALLOW_REFRESH_TOKEN_AUTH"
+];
+// highlight-end
+```
+
+### Migrate users with Amazon Cognito
```

Comment:
can we reuse the prose from other platforms here? (aside from the platform-specific flow enums). ideally we retain the existing prose to reduce maintenance overhead

Reply:
Can you point me to the other platform's section you want me to emulate? The way that I see it, this matches 1:1 with Swift and there's a slight difference with the last sentence with JS:

Android and Swift:

> There's documentation around [how to set up this migration flow](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-import-using-lambda.html) and more detailed instructions on [how the lambda should handle request and response objects](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html#cognito-user-pools-lambda-trigger-syntax-user-migration).

JS:
> Visit [Amazon Cognito user pools import guide](https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-user-pools-import-using-lambda.html) for migration flow and more detailed instruction, and [Amazon Cognito Lambda trigger guide](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-migrate-user.html#cognito-user-pools-lambda-trigger-syntax-user-migration) on how to set up lambda to handle request and response objects.

Reply:
ah okay, we can refactor this later, then